agent/grpc-external/services/resource/delete_test.go

```diff
@@ -140,10 +142,64 @@ func TestDelete_Success(t *testing.T) {
 			_, err = server.Backend.Read(ctx, storage.StrongConsistency, artistId)
 			require.Error(t, err)
 			require.ErrorIs(t, err, storage.ErrNotFound)
+
+			// verify tombstone created
+			stream, err := client.WatchList(ctx, &pbresource.WatchListRequest{
+				Type:       resource.TypeV1Tombstone,
+				Tenancy:    artist.Id.Tenancy,
+				NamePrefix: "",
+			})
+			require.NoError(t, err)
+			rspCh := handleResourceStream(t, stream)
+			event := mustGetResource(t, rspCh)
+			require.Equal(t, event.Operation, pbresource.WatchEvent_OPERATION_UPSERT)
+			tombstone := &pbresource.Tombstone{}
+			require.NoError(t, event.Resource.Data.UnmarshalTo(tombstone))
+			prototest.AssertDeepEqual(t, artistId, tombstone.OwnerId)
+			require.Equal(t, artist.Version, tombstone.OwnerVersion)
```

Comment:
Is there a particular reason to use `WatchList` here, rather than `Read`-ing the tombstone?

Reply:
Yup, way simpler! https://github.com/hashicorp/consul/pull/17108/commits/ea8dc796d36d5f1b9aef7eddac7564573437d798